File: Blog.Services/RecordDTO.cs
using Blog.Services.Enums;
using Blog.Services.Identity;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Blog.Services
{
    public class RecordDTO
    {
        public int RecordId { get; set; }

        [Required(ErrorMessage = "Please, set Name")]
        [StringLength(50, MinimumLength = 5)]
        public string Name { get; set; }

        [Required(ErrorMessage = "Please, set Content")]
        [StringLength(1000, MinimumLength = 5)]
        public string Content { get; set; }

        public DateTime CreateDate { get; set; }

        public RecordStateDTO State { get; set; } = RecordStateDTO.Private;

        public ICollection<CommentDTO> Comments { get; set; } = new List<CommentDTO>();

        public string UserId { get; set; }

        public UserDTO User { get; set; }
    }
}

File: Blog.Site/Scripts/react/Src/Types/RecordDTO.ts
import RecordStateDTO from "types/RecordStateDTO";
import UserDTO from "types/UserDTO";
import CommentDTO from "types/CommentDTO";

export default class RecordDTO {
  public RecordId: number;

  public Name: string;

  public Content: string;

  public CreateDate: Date;

  public State: RecordStateDTO;

  public Comments: CommentDTO[];

  public UserId: string;

  public User: UserDTO;
}
